plugins/tableselection/plugin.js

```diff
@@ -251,7 +252,7 @@
 
 		if ( cells.length > 0 ) {
 			editor.editable().addClass( fakeSelectedEditorClass );
-			cells[ 0 ].getAscendant( 'table' ).addClass( fakeSelectedTableClass );
+			cells[ 0 ].getAscendant( 'table' ).data( fakeSelectedTableDatasetClass, fakeSelectedTableClass );
```

Comment:
Makes sense 👍 .
It is similar behaviour to how `Boolean attributes` works (e.g. `input[checked]`). Here we also have only two values, `selected` / `not selected` so it is a proper approach IMHO.